app/gui/src/controller/ide/desktop.rs

```diff
@@ -167,11 +167,22 @@ impl ManagingProjectAPI for Handle {
 
 /// Select a new name for the project in a form of <suggested_name>_N, where N is a unique sequence
 /// number.
-fn choose_new_project_name(existing_names: &HashSet<String>, suggested_name: &str) -> String {
+fn choose_unique_project_name(existing_names: &HashSet<String>, suggested_name: &str) -> String {
     let first_candidate = suggested_name.to_owned();
     let nth_project_name = |i| iformat!("{suggested_name}_{i}");
     let candidates = (1..).map(nth_project_name);
     let mut candidates = iter::once(first_candidate).chain(candidates);
     // The iterator have no end, so we can safely unwrap.
     candidates.find(|c| !existing_names.contains(c)).unwrap()
 }
+
+/// Come up with a project name.
+fn make_project_name(template: &Option<String>) -> String {
+    let mut name = template.clone().unwrap_or_else(|| UNNAMED_PROJECT_NAME.to_owned());
+    // Capitalize
+    if let Some(r) = name.get_mut(0..1) {
+        r.make_ascii_uppercase();
+    }
+
+    name
+}
```

Comment:
Are we sure this check is enough?

We take the template name, which, as I understand, could be just _anything_, contains spaces dollars etc. And if it could not, and we ensure somewhere that there are no such characters there, why not to check the casing in the same place?

Reply:
I guess, I can try making a newtype wrapper for template names, that will check that they only contain accepted characters

Reply:
I updated the PR. Now there is a `Template` wrapper ensuring that the template name is valid